### PR TITLE
Add FlapperUniV2SwapOnly

### PR DESCRIPTION
**Pull request by sunbreak1211**

### CODE REVIEW

File: src/FlapperUniV2.sol [hunk 1]
```diff
@@ -35,7 +35,6 @@ interface SpotterLike {
 interface GemLike {
     function decimals() external view returns (uint8);
     function balanceOf(address) external view returns (uint256);
-    function approve(address, uint256) external;
```
Comment (oldchili) [inline src/FlapperUniV2.sol:38]: If we want to remove unused staff then we can also remove the `live()` from `VatLike`.
I'm not 100% sure these are worth changing that contract and differing from the deployed version.

File: src/FlapperUniV2SwapOnly.sol [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    function _getReserves() internal returns (uint256 reserveDai, uint256 reserveGem) {
+        (uint256 _reserveA, uint256 _reserveB,) = pair.getReserves();
+        (reserveDai, reserveGem) = daiFirst ? (_reserveA, _reserveB) : (_reserveB, _reserveA);
+
+        uint256 _daiBalance = GemLike(dai).balanceOf(address(pair));
+        uint256 _gemBalance = GemLike(gem).balanceOf(address(pair));
+        if (_daiBalance > reserveDai || _gemBalance > reserveGem) {
+            pair.sync();
+            (reserveDai, reserveGem) = (_daiBalance, _gemBalance);
+        }
```
Comment (oldchili) [inline src/FlapperUniV2SwapOnly.sol:145]: We should consider removing this.

File: src/FlapperUniV2SwapOnly.sol [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+interface PipLike {
+    function read() external view returns (bytes32);
+}
+
+// https://github.com/Uniswap/v2-core/blob/ee547b17853e71ed4e0101ccfd52e70d5acded58/contracts/UniswapV2Pair.sol
+interface PairLike {
+    function getReserves() external view returns (uint112 reserve0, uint112 reserve1, uint32 blockTimestampLast);
+    function token0() external view returns (address);
+    function mint(address to) external returns (uint256 liquidity);
```
Comment (oldchili) [inline src/FlapperUniV2SwapOnly.sol:47]: `mint` can also be removed.

File: src/FlapperUniV2SwapOnly.sol [hunk 4]
```diff
@@ -0,0 +1,183 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
```
Comment (oldchili) [inline src/FlapperUniV2SwapOnly.sol:1]: When finalizing we need to add this contract to the README.

File: src/FlapperUniV2SwapOnly.sol (hunk 2 above)
Comment (sunbreak1211) [reply src/FlapperUniV2SwapOnly.sol]: Done.

File: src/FlapperUniV2SwapOnly.sol (hunk 3 above)
Comment (sunbreak1211) [reply src/FlapperUniV2SwapOnly.sol]: Done.

File: src/FlapperUniV2SwapOnly.sol (hunk 4 above)
Comment (sunbreak1211) [reply src/FlapperUniV2SwapOnly.sol:1]: Done.

File: README.md [hunk 5]
```diff
@@ -11,6 +11,15 @@ Configurable Parameters:
 * `pip` - A reference price oracle, used for bounding the exchange rate of the swap.
 * `want` - Relative multiplier of the reference price to insist on in the swap. For example, a value of 0.98 * `WAD` allows for a 2% worse price than the reference.
 
+### FlapperUniV2SwapOnly
```
Comment (oldchili) [inline README.md:14]: Be aware that there are notes at the end that are only relevant to the current flapper.
Maybe we should just change "also withdraws"  to "may also withdraw".

Comment (sunbreak1211) [reply README.md:20]: Moved things a bit around, let me know what you think.

Comment (oldchili) [reply README.md:20]: Yeah ok, this was the other alternative I had in mind.
Works as well.

**Review by oldchili — APPROVED** [review]
LGTM

**Review by oldchili — APPROVED** [review]
Last small change also good.

File: test/FlapperUniV2SwapOnly.t.sol [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+    address     MKR;
+    address     USDC;
+    address     LINK;
+    address     PAUSE_PROXY;
+    VatLike     vat;
+    VowLike     vow;
+    EndLike     end;
+    SpotterLike spotter;
+
+    address     constant  LOG           = 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F;
```
Comment (telome) [inline test/FlapperUniV2SwapOnly.t.sol:109]:
```suggestion
    address constant LOG                 = 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F;
```

Comment (telome) [reply test/FlapperUniV2SwapOnly.t.sol:109]: (And the same nit in test/FlapperUniV2.t.sol)

**Review by telome — DISMISSED** [review]
LGTM!

Just a small note regarding the removal of the insanity check: it could be argued that in addition to protecting the SB from being drained in a scenario of low liquidity + faulty oracle and/or incorrectly configured `want`, the check was also setting a hard cap on the swap slippage under these failure modes. So technically we are losing this protection here but I don't think it's a problem as the `hop` & `lot` combo makes sure that such failure can only have limited impact before it can be caught and mitigated.

File: test/FlapperUniV2SwapOnly.t.sol (hunk 6 above)
Comment (sunbreak1211) [reply test/FlapperUniV2SwapOnly.t.sol]: Fixed in both files!

**Review by oldchili — APPROVED** [review]
LGTM

**Pull request merged**